kbase-extension/static/kbase/js/common/jobs.js

```diff
@@ -1,6 +1,205 @@
+     *
+     * @param {string} jobState
+     * @returns {string} label
+     */
+
+    function jobLabel(jobState) {
+        // covers 'does_not_exist' or invalid job states
+        let label = 'Job not found';
+        switch (jobState) {
+            case 'completed':
+                label = 'Success';
+                break;
+            case 'created':
+            case 'estimating':
+            case 'queued':
+                label = 'Queued';
+                break;
+            case 'running':
+                label = 'Running';
+                break;
+            case 'error':
+                label = 'Failed';
+                break;
+            case 'terminated':
+                label = 'Cancelled';
+                break;
+        }
+        return label;
+    }
+
+    /**
+     * Translate from EE2's job status (or other job state strings interpreted by the app cell)
+     * to a presentable string. This returns a span with the text colored and bolded, and the
+     * "nice" readable state string.
+     *
+     * Translated strings = completed, error, terminated, and does_not_exist. Those all get
+     * different colors. Other strings are rendered black.
+     * @param {string} jobState
+     */
+
+    function niceState(jobState) {
+        let label,
+            color;
+        switch (jobState) {
+            case 'completed':
+                label = 'success';
+                color = 'green';
+                break;
+            case 'error':
+                label = 'error';
+                color = 'red';
+                break;
+            case 'terminated':
+                label = 'cancellation';
+                color = 'orange';
+                break;
+            case 'does_not_exist':
+                label = 'does_not_exist';
+                color = 'orange';
+                break;
+            default:
+                label = jobState;
+                color = 'black';
+        }
+
+        return span({
+            class: `${cssBaseClass}__niceState--${jobState}`,
+            style: {
+                color: color,
+                fontWeight: 'bold'
+            }
+        }, label);
+    }
+
+    function jobNotFound() {
+        return [
+            'This job was not found, or may not have been registered with this Narrative.',
+            'You will not be able to inspect the job status or view the job log'
+        ];
+    }
+
+    function jobStatusUnknown() {
+        return ['Determining job state...'];
+    }
+
+    /**
+     *
+     * @param {object}  jobState       - object representing the current job state
+     * @param {boolean} includeHistory - whether to generate a single line status (false), or
+     *                               to include the history (e.g. how long the job queued for)
+     * @returns {array} jobLines       - an array of lines representing the current job status
+     */
+
+    function createJobStatusLines(jobState, includeHistory = false) {
```

Comment:
The logic (order of `if`s, array reversing) in this function seems a bit overly complicated (at least I had a hard time reading it). Could maybe be simplified, though since its well tested it doesn't seem like a high risk place to have complexity. 🤷

Reply:
I've added some more documentation to explain what's going on. Having been deeply embedded in job statuses for the past week, it's easy to forget that not everyone is lucky enough to have this much knowledge of the various states...